common/hyperscaler/rules/attribute.go

```diff
@@ -0,0 +1,35 @@
+package rules
+
+import "fmt"
+
+type Attribute struct {
+	Name            string
+	Description     string
+	Setter          func(*Rule, string) (*Rule, error)
+	Getter          func(*Rule) string
+	MatchableGetter func(*MatchableAttributes) string
+	input           bool
+	output          bool
+	modifiedLabel   string
+	HasValue        bool
+
+	modifiedLabelName string
+	ApplyLabel        func(r *Rule, labels map[string]string) map[string]string
+}
+
+func (a Attribute) HasLiteral(rule *Rule) bool {
+	value := a.Getter(rule)
+	return value != "*" && value != ""
+}
+
+func (a Attribute) String(r *Rule) any {
+	val := a.Getter(r)
```

Comment:
Why do we use string representation of boolean?
It seems that Getter can return:
- "true"
- "false"
- ""
- "*"
- something else

Reply:
For convenience.

Reply:
Not the reader/maintainer convenience.